libcxx/test/support/test_iterators.h

```diff
@@ -1114,6 +1114,23 @@ rvalue_iterator(T*) -> rvalue_iterator<T>;
 
 static_assert(std::random_access_iterator<rvalue_iterator<int*>>);
 
+// The ProxyDiffTBase allows us to conditionally specify Proxy<T>::difference_type
+// which we need in certain situations. For example when we want
+// std::weakly_incrementable<Proxy<T>> to be true.
+template <class T>
+struct ProxyDiffTBase {
+  // Add default `operator<=>` so that the derived type, Proxy, can also use the default `operator<=>`
+  friend constexpr auto operator<=>(const ProxyDiffTBase&, const ProxyDiffTBase&) = default;
```

Comment:
Nit: pass by value (same elsewhere for types that you wouldn't ordinarily pass by ref-to-const).

Reply:
I agree it seems a bit unnecessary to pass by ref-to-const here, but I was basing these on the existing `Proxy<T>` which uses ref-to-const